resource/menu/shared.lua

```diff
@@ -26,6 +26,20 @@ function txPrint(...)
   print(msg)
 end
 
+local supressedEvents = GetConvar('txAdmin-supressEvents', '')
+
+-- Used to check if a specific event should be suppressed
+function isSuppressedEvent(eventName)
+  for eventToBypass in string.gmatch(supressedEvents, '[^,%s]+') do
+      if string.lower(eventName) == string.lower(eventToBypass) then
+          return true
+      end
+  end
+
+  return false
+end
```

Comment:
It is possible to slightly reduce time complexity by splitting and transforming the ConVar string into a map table. Then, once in the function, index table against passed eventName using string index on map table. Reduces time complexity at the cost of space complexity.

If what I said makes no sense, don't worry about it, and keep it as is. This isn't a huge deal at all, and actual real-world use cases will see minimal difference. Change is up to you.

Reply:
Sounds good. I will adjust it that way.